### ISSUE TITLE
environment config file does not include channels from user config file

### ISSUE THREAD

**Issue by johnholman** [conda/conda#11793]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

In general I want the channel list to be conda-forge then defaults. However for some environments (e.g. pytorch) I want the pytorch channel to be searched first, and so want to create a suitable environment-specific config file

I started with a user .condarc file with contents

channel_priority: strict
channels:
  - conda-forge
  - defaults

(no system config file or any other config sources)

Then added an environment config file with

conda config --env --add channels pytorch

But channels in the user config file seem to be ignored resulting in environment config file 

channel_priority: strict
channels:
  - pytorch
  - defaults

As a result the defaults channel gets priority over conda-forge, which is not what I want:

(pytorch) C:\Users\johng>conda config --show channels
channels:
  - pytorch
  - defaults
  - conda-forge

I think this may be a bug, as it works as I expected if I make the initial config file a system config file rather than a user one. With this workaround i get the required environment config file:

channel_priority: strict
channels:
  - pytorch
  - conda-forge
  - defaults

### Conda Info

```shell
(base) C:\Users\johng>conda info

     active environment : base
    active env location : C:\Users\johng\miniconda3
            shell level : 1
       user config file : C:\Users\johng\.condarc
 populated config files : C:\Users\johng\miniconda3\.condarc
          conda version : 4.14.0
    conda-build version : not installed
         python version : 3.9.13.final.0
       virtual packages : __cuda=11.7=0
                          __win=0=0
                          __archspec=1=x86_64
       base environment : C:\Users\johng\miniconda3  (writable)
      conda av data dir : C:\Users\johng\miniconda3\etc\conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\johng\miniconda3\pkgs
                          C:\Users\johng\.conda\pkgs
                          C:\Users\johng\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\johng\miniconda3\envs
                          C:\Users\johng\.conda\envs
                          C:\Users\johng\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.14.0 requests/2.28.1 CPython/3.9.13 Windows/10 Windows/10.0.22000
          administrator : False
             netrc file : None
           offline mode : False
```

### Conda Config

```shell
(pytorch) C:\Users\johng>conda config --show-sources
==> C:\Users\johng\.condarc <==
channel_priority: strict
channels:
  - conda-forge
  - defaults

==> C:\Users\johng\miniconda3\envs\pytorch\.condarc <==
channels:
  - pytorch
  - defaults
```

### Conda list

```shell
(pytorch) C:\Users\johng>conda list --show-channel-urls
# packages in environment at C:\Users\johng\miniconda3\envs\pytorch:
#
# Name                    Version                   Build  Channel
```

**Comment by travishathaway** [comment]
Hi @johnholman,

It sounds like you may attempting to do something for which there exists a solution: `conda env`.  You can read more about it here:
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file

**TLDR**: basically `conda env` allows you to use environment specific config files. So, it's possible to define an environment which looks like this:

```yaml
# environment.yml
name: my_pytorch_env
channels:
  - pytorch
  - defaults
dependencies:
  - python
  - pytorch
```

And then install it like so:

```bash
$ conda env create -f environment.yml
```

Let us know if you need more help.
